src/Features/LanguageServer/Protocol/Handler/PullHandlers/VersionedPullCache`2.cs

```diff
@@ -38,7 +38,7 @@ internal class VersionedPullCache<TCheapVersion, TExpensiveVersion>
         /// </list>
         /// This is used to determine if we need to re-calculate results.
         /// </summary>
-        private readonly Dictionary<(Workspace workspace, DocumentId documentId), (string resultId, TCheapVersion cheapVersion, TExpensiveVersion expensiveVersion)> _documentIdToLastResult = new();
+        private readonly Dictionary<(Workspace workspace, DocumentId documentId), (string resultId, int encVersion, TCheapVersion cheapVersion, TExpensiveVersion expensiveVersion)> _documentIdToLastResult = new();
```

Comment:
Alternate approach here - make TCheap version a tuple or something that contains both the version stamp and the enc version, then you can just leave the cache as is - which seems a little cleaner

Reply:
Great idea! I guess comparing ints could probably be called cheap :)

Reply:
Okay, its slightly less great than I originally hoped, because both version checks need to become tuples, but I like that the cache is left alone. Not sure 🤷‍♂️

Reply:
This definitely looks better to me - but why did the version checks both need to become tuples?  TCheapVersion and TExpensiveVersion aren't related

Reply:
Yes, sadly. The cache checks the fast check, and then the slow check, but then uses the cached result if either of them match. Without the slow check doing EnC stuff, it matches and we don't get new diagnostics.

Reply:
Had to update the cache anyway.. it was doing `??=` which of course stops working when you change the generic type to a struct 🤦‍♂️

Still not as bad a change as introducing the concept of EnC though :)